### PR TITLE
workflows: windows: Implement cache mechanism for source packages of vcpkg

### PR DESCRIPTION
**Pull request by cosmo0920**
Currently, nasm.us is occasionally down.
To improve our CI stability, we have to implement caching mechanism for source packages to stabilize vcpkg installations.

CI result is here: https://github.com/fluent/fluent-bit/actions/runs/5710860698

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [ ] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [x] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [ ] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: .github/workflows/call-build-windows.yaml [hunk 1]
```diff
@@ -112,6 +112,27 @@ jobs:
...
+      - name: Restore cached packages of vcpkg
+        id: cache-vcpkg-sources
+        uses: actions/cache/restore@v3
+        with:
+          path: |
+            C:\vcpkg\packages
+          key: ${{ runner.os }}-${{ matrix.config.arch }}-vcpkg
+          enableCrossOsArchive: false
+
       - name: Build openssl with vcpkg
```
Comment (patrick-stephens) [inline .github/workflows/call-build-windows.yaml:136]: Do we need to skip on cache hit?

File: .github/workflows/call-build-windows.yaml [hunk 2]
```diff
@@ -112,6 +112,27 @@ jobs:
         with:
           arch: ${{ matrix.config.arch }}
 
+      - name: Get gzip command w/ chocolatey
+        uses: crazy-max/ghaction-chocolatey@v2
+        with:
+          args: install gzip -y
```
Comment (patrick-stephens) [inline .github/workflows/call-build-windows.yaml:118]: What's the reason for needing gzip?

File: .github/workflows/call-build-windows.yaml (hunk 1 above)
Comment (cosmo0920) [reply .github/workflows/call-build-windows.yaml:136]: No. We're only caching for sources of openssl and libyaml. We always need to build them on the every run.

File: .github/workflows/call-build-windows.yaml (hunk 2 above)
Comment (cosmo0920) [reply .github/workflows/call-build-windows.yaml:118]: The reason is `actions/cache/save@v3` is heavily depends on gzip command actually...

**Pull request merged**